dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCFile.java

```diff
@@ -422,8 +422,17 @@ private void createDataParts() throws HpccFileException
                 }
 
                 this.recordDefinition = RecordDefinitionTranslator.parseJsonRecordDefinition(new JSONObject(originalRecDefInJSON));
-                
-                this.partitionProcessor = new PartitionProcessor(this.recordDefinition, this.dataParts, this.tlkPartition);
+
+                try
+                {
+                    this.partitionProcessor = new PartitionProcessor(this.recordDefinition, this.dataParts, this.tlkPartition);
+                }
+                catch (Exception e)
+                {
+                    log.error("Error while constructing partition processor, reading will continue without partition filtering: " + e.getMessage());
+                    this.partitionProcessor = new PartitionProcessor(this.recordDefinition, this.dataParts, null);
+                }
```

Comment:
any way additional info could be logged (within the BinaryRecordReader probably) to pinpoint the field causing the error, and even the record causing the error?)